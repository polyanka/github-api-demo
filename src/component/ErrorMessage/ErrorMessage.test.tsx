import { mount, shallow } from 'enzyme';
import * as React from 'react';

import { ErrorMessage } from './index';

describe('<ErrorMessage /> ', () => {
  it('renders correctly ErrorMessage component', () => {
    const wrapper = shallow(<ErrorMessage />);
    expect(wrapper).toMatchSnapshot();
  });

  it('check prop message by default', () => {
    const wrapper = mount(<ErrorMessage />);
    expect(wrapper.find('h5').text()).toEqual('Typo An Error Occurred, Please Try Again Later!');
  });

  it('check prop message', () => {
    const wrapper = mount(<ErrorMessage message='Error!' />);
    expect(wrapper.find('h5').text()).toEqual('Error!');
  });
});
