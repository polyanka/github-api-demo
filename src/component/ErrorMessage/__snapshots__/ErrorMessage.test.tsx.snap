// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage />  renders correctly ErrorMessage component 1`] = `
<WithStyles(ForwardRef(Grid))
  item={true}
>
  <WithStyles(ForwardRef(Typography))
    color="error"
    variant="h5"
  >
    Typo An Error Occurred, Please Try Again Later!
  </WithStyles(ForwardRef(Typography))>
</WithStyles(ForwardRef(Grid))>
`;
